first-pr-sample (2021)

sofyanox12 opened the pull request:
👍

sofyanox12 committed:
Update Cashier.java

sofyanox12 committed:
Update RoleAccess.java

fitrh reviewed (changes requested):
Perhatikan [spesifikasi][spec] dan [video referensi][video-ref], dan pastikan program anda berjalan sama atau lebih baik dari apa yang ditampilkan pada [video referensi][video-ref] dan memnuhi semua poin-poin [spesifikasi][spec]

[spec]: https://gist.github.com/fitrh/f6faadd5a967c9d270fa3e73dd42b034#spesifikasi
[video-ref]: https://gist.github.com/fitrh/f6faadd5a967c9d270fa3e73dd42b034#referensi

sofyanox12 committed:
Revisi Cashier.java (assignment 1.1)

Penentuan variabel sudah diubah barisannya dan lebih deskiriptif, serta pada interface hanya akan menerima inputan dalam bentuk integer sesuai arahan kakak.

sofyanox12 committed:
Revisi RoleAccess.java (Assignment 1.2)

Maaf kak, yang bisa saya pikirkan hanya beberapa pada method untuk "args[0]" dalam kasus ini kak, sekiranya bgtumi kak.

sofyanox12 committed:
Update Cashier.java

sofyanox12 committed:
Update RoleAccess.java

sofyanox12 committed:
Update : Change money Mechanism

bug : Kembalian yang nyata untuk pecahan kecil harga dengan donasi yang terlalu besar.
fix : Menambahkan objek pecahan untuk setiap kondisi pengembalian.

sofyanox12 committed:
add

sofyanox12 committed:
Delete Eyes.java

sofyanox12 committed:
Delete MatricesPad.java

sofyanox12 committed:
Delete FactorialSequence.java

sofyanox12 committed:
add MatricesPad.java

sofyanox12 committed:
Update MatricesPad.java

Note fixed

sofyanox12 committed:
add FactorialSequence.java

sofyanox12 committed:
add Eyes.java

sofyanox12 committed:
Update Eyes.java

Memperbaiki elemen spasi pada setiap barisan dan diakhiri format newLine pada akhir baris.

sofyanox12 committed:
Update Eyes.java

fitrh reviewed (approved):
Jangan abai persoalan format white space

sofyanox12 commented on src/main/java/ip/syssrc/loop/Eyes.java:
```diff
@@ -0,0 +1,53 @@
+                } else {
+                    System.out.print("0");
+                }
+                if (t==(number-1)){
```
baik kak

fitrh commented on src/main/java/ip/syssrc/loop/MatricesPad.java:
```diff
@@ -0,0 +1,70 @@
+            
+            // Line output by selections
+            if ((i != start) && ((i - start) % size == 0 || (start - i) % size == 0)) {
+                System.out.println(" ");
```
Baris terakhir merupakan *new line* tanpa *space*

fitrh commented on src/main/java/ip/syssrc/loop/MatricesPad.java:
```diff
@@ -0,0 +1,70 @@
+                System.out.println(" ");
+            }
+
+            // Numbers output
+            System.out.printf("%0" + d + "d ", i);
```
Elemen terakhir diakhiri dengan *new line*, bukan *space*

**Contoh Input**

```
0 10 3
```

**Contoh Output**

```
00 01 02
03 04 05
06 07 08
09 10
```

Jika divisualisasikan, baris terakhir seperti berikut

```txt
09<space>10<newline>
```

Sedangkan kode ini akan menghasilkan output berikut

```txt
09<space>10<space>
```

sofyanox12 committed:
Update MatricesPad.java

fix the previous spacing and newlines printout

sofyanox12 committed:
Update Eyes.java

fitrh commented:
2 commit terakhir belum memperbaiki review yang saya berikan

sofyanox12 committed:
Update MatricesPad.java : space and newlines printout fix

sofyanox12 committed:
Update Eyes.java : fix the newlines and spaces on the output

fitrh commented on src/main/java/ip/syssrc/loop/MatricesPad.java:
```diff
@@ -0,0 +1,74 @@
+
+            // Spacing or Newlines detector
+            if (i == end || (opt != 0 && opt % size == 0)) {
+            System.out.print("\n");
+            } else {
+            System.out.print(" ");
```
Jangan mengabaikan penggunan indentasi

fitrh commented on src/main/java/ip/syssrc/loop/MatricesPad.java:
```diff
@@ -0,0 +1,74 @@
+                ++i;
+            }
+        }
+        System.out.println("\n");
```
Baris ini sudah tidak diperlukan

sofyanox12 committed:
Update MatricesPad.java

fitrh commented on src/main/java/ip/syssrc/loop/MatricesPad.java:
```diff
@@ -0,0 +1,73 @@
+        for (int i = start; (start < end ? i <= end : i >= end); ) {
+
+            // Numbers output
+            System.out.printf("%0" + d + "d", i);
+
+            // Spacing or Newlines detector
+            if (i == end || (opt != 0 && opt % size == 0)) {
+                System.out.print("\n");
```
Baris ini akan mengashilkan *exception* untuk input berikut

```
0 0 1
```

```
0 0 0
```

Untuk kedua input diatas, program menghasilkan output 0

sofyanox12 replied:
kak, kalau untuk input start dan end nya sama itu outputnya juga 0?

fitrh replied:
Tidak, kalau sama, berarti outputnya angka tersebut

```
10 10 0
```

atau

```
10 10 1
```

Outputnya

```
10
```

sofyanox12 committed:
Update MatricesPad.java : Adding the exception filter for some conditions.

sofyanox12 committed:
Update MatricesPad.java

fitrh commented on src/main/java/ip/syssrc/loop/MatricesPad.java:
```diff
@@ -0,0 +1,85 @@
+        System.out.println("\nResult :");
+        for (int i = start; (start < end ? i <= end : i >= end); ) {
+
+            if (start == end) { // The loop will stop if start = end
+                System.out.println(start - end + "\n\n[Keterangan!] Start dan End berhimpit.\n");
+                break;
+            }
```
Tidak pernah ada syarat ini, tetap mencetak angka tersebut

sofyanox12 replied:
ohh baik kak.

sofyanox12 committed:
Update MatricesPad.java : change the output of start = end.

sofyanox12 committed:
Update MatricesPad.java : change the condition of exception filter
